src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,279 @@
+﻿/* @internal */
+namespace ts.codefix {
+    const nodeModulesFolderName = "node_modules";
+
+    registerCodeFix({
+        errorCodes: [Diagnostics.Cannot_find_name_0.code],
+        getCodeActions: (context: CodeFixContext, cancellationToken: CancellationToken) => {
+            const sourceFile = context.sourceFile;
+            const checker = context.program.getTypeChecker();
+            const allSourceFiles = context.program.getSourceFiles();
+            const useCaseSensitiveFileNames = context.host.useCaseSensitiveFileNames ? context.host.useCaseSensitiveFileNames() : false;
+
+            const token = getTokenAtPosition(sourceFile, context.span.start);
+            const name = token.getText();
+            const allActions: CodeAction[] = [];
+            let allPotentialModules: Symbol[] = [];
+
+            const ambientModules = checker.getAmbientModules();
+            if (ambientModules) {
+                allPotentialModules = ambientModules;
+            }
+            for (const otherSourceFile of allSourceFiles) {
+                if (otherSourceFile !== sourceFile && otherSourceFile.symbol) {
+                    allPotentialModules.push(otherSourceFile.symbol);
+                }
+            }
+
+            for (const moduleSymbol of allPotentialModules) {
+                cancellationToken.throwIfCancellationRequested();
+
+                const exports = checker.getExportsOfModule(moduleSymbol) || [];
+                for (const exported of exports) {
+                    if (exported.name === name) {
+                        allActions.push(getCodeActionForImport(moduleSymbol));

```

Comment:
We also need to handel the default export correctelly:


```ts
 if (result = moduleExports["default"]) {
    const localSymbol = getLocalSymbolForExportDefault(result);
    if (localSymbol && (result.flags & meaning) && localSymbol.name === name) {
        // found it
    }
 // otherwise it is not the right match
}               
```


```ts
// a.ts

Foo

// b.ts
export default class Foo {
}   
```